db/migrate/20240904141112_update_fa5_icon_names.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class UpdateFa5IconNames < ActiveRecord::Migration[7.1]
+  def up
+    migrate_value("up")
+  end
+
+  def down
+    migrate_value("down")
+  end
+
+  def migrate_value(dir)
+    icons =
+      File.open("#{Rails.root}/db/migrate/20240904140851_fa5_renames.json", "r:UTF-8") do |f|
+        JSON.parse(f.read)
+      end
+
+    icons.each do |key, value|
+      from = dir == "up" ? key : value
+      to = dir == "up" ? value : key
+      execute <<~SQL
+        UPDATE badges
+        SET icon = '#{to}'
+        WHERE icon = '#{from}' OR icon = 'fa-#{from}'
```

Comment:
IIRC the `fa-` prefix was something that was only on FA4. With FA5 onwards, however, we have the `far-` and `fab-` prefixes, which are Discourse-only constructs. They probably are unchanged between 5 and 6, but since this is a migration messing with DB data (and therefore not easy to undo), it might be worth double-checking that any `far-` or `fab-` prefixes work fine before/after the migration.